middleware/proxy.go

```diff
@@ -287,19 +302,52 @@ func ProxyWithConfig(config ProxyConfig) echo.MiddlewareFunc {
 				req.Header.Set(echo.HeaderXForwardedFor, c.RealIP())
 			}
 
-			// Proxy
-			switch {
-			case c.IsWebSocket():
-				proxyRaw(tgt, c).ServeHTTP(res, req)
-			case req.Header.Get(echo.HeaderAccept) == "text/event-stream":
-			default:
-				proxyHTTP(tgt, c, config).ServeHTTP(res, req)
-			}
-			if e, ok := c.Get("_error").(error); ok {
-				err = e
-			}
+			retries := config.RetryCount
+			for {
+				var tgt *ProxyTarget
+				var err error
+				if isTargetProvider {
+					tgt, err = provider.NextTarget(c)
+					if err != nil {
+						return err
+					}
+				} else {
+					tgt = config.Balancer.Next(c)
+				}
+				c.Set(config.ContextKey, tgt)
```

Comment:
Clearing error here would have some benefits
```go
				c.Set(config.ContextKey, tgt)
				if retries < config.RetryCount {
					// error is cleared where so on retry we could check err in `provider.NextTarget(c)` call and decide next
					// target based on that error
					c.Set("_error", nil)
				}
```

Reply:
You would still like to keep this - this allows user to build `Balancer` implementations that can decide things from err and previous target.